### PR TITLE
Implement Opening a FoodStorage Object

### PR DESCRIPTION
**Pull request by amandate**
# Issue/Task:
Implement logic for opening a `FoodStorage` object.

When we do `FoodStorage.open()` we expect:
* `isOpen` gets set to `True`
* `current_date` get updated to today's date
* we list the current state of all the foods inside the `FoodStorage` object

This PR also handles some clean up tasks for how print output looks for `list()` and `addFoods()`
We want to add a little notice of foods that are:
* expired
* about to expire within the set window

# Checklist
- [x] Fixed Issue/Accomplished Task
- [x] Wrote tests
- [x] All test cases passing

### CODE REVIEW

File: src/Constants/constants.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ADD_FOODS_SUCCESS_MESSAGE = "Successfully added all foods: {}"
```
Comment (leezahuor) [inline src/Constants/constants.py:1]: Could this file be named something else (like "messages.py" or something) since there's already a "constants.py"?

File: src/FoodStorage/foodStorage.py [hunk 2]
```diff
@@ -1,16 +1,30 @@
+from src.Constants.constants import \
+    ADD_FOODS_SUCCESS_MESSAGE, \
```
Comment (leezahuor) [inline src/FoodStorage/foodStorage.py:2]: Why did you have to import every constant individually?

File: src/FoodStorage/foodStorage.py [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+    FOOD_ABOUT_TO_EXPIRE_MESSAGE_SINGULAR, \
+    FOOD_EXPIRED_MESSAGE, \
+    FOOD_EXPIRED_MESSAGE_SINGULAR, \
+    FOOD_EXPIRED_TODAY_MESSAGE
+from datetime import date, timedelta
 from queue import PriorityQueue
 
 class FoodStorage:
-    def __init__(self):
+    def __init__(self, expiration_window=7):
```
Comment (leezahuor) [inline src/FoodStorage/foodStorage.py:13]: Why did you add `expiration_window=7`?

File: tests/FoodStorage/test_foodStorage.py [hunk 4]
```diff
@@ -1,14 +1,19 @@
-import unittest
+from src.Constants.constants import *
 from src.FoodStorage.foodStorage import FoodStorage
 from tests.Utilities.constants import *
 from tests.Utilities.utils import *
 
+import io
+import sys
```
Comment (leezahuor) [inline tests/FoodStorage/test_foodStorage.py:7]: How is `io` and `sys` being used in this test?

File: tests/FoodStorage/test_foodStorage.py [hunk 5]
```diff
@@ -22,12 +27,48 @@ def testAddFoods(self):
...
+            )
+
+    def testList(self):
+        self.assertTrue(isEmptyInventory(self.foodStorage))
+
+        self.foodStorage.addFoods(
+            [foodFuture, foodTomorrow, foodNearFuture, foodToday, foodPast, foodYesterday])
+
+        # Grab print output from calling list()
+        capturedListOutput = io.StringIO()
```
Comment (leezahuor) [inline tests/FoodStorage/test_foodStorage.py:45]: What is happening here?

**Review by leezahuor — COMMENTED** [review]
Just need a bit of clarification. Thanks!

File: src/FoodStorage/foodStorage.py (hunk 2 above)
Comment (amandate) [reply src/FoodStorage/foodStorage.py:2]: So I guess for here, it looks like it doesn't make sense to import everything individually because we're using everything. But if we were adding more things to the constants file and only using a part of it here, we only want to import the things we use. I guess I can change it to import all for now though and we can change it to individual variables if we end up adding more to the constants file.

File: src/FoodStorage/foodStorage.py (hunk 3 above)
Comment (amandate) [reply src/FoodStorage/foodStorage.py:13]: The `expiration_window` represents the window for what foods we want to display a notice for that are about to expire. I thought a default of 7 would be good. Basically this means when we create a `FoodStorage` object, by default, we give notice of foods that are going to expire in 7 days.

File: tests/FoodStorage/test_foodStorage.py (hunk 4 above)
Comment (amandate) [reply tests/FoodStorage/test_foodStorage.py:7]: We use it to grab the print output. I wanted to test that our print statements were displaying correctly for different cases.

File: tests/FoodStorage/test_foodStorage.py (hunk 5 above)
Comment (amandate) [reply tests/FoodStorage/test_foodStorage.py:45]: This is how we grab the result of printing things to standout. I'm basically grabbing the print statement so I can test that it prints as expected below.

File: src/Constants/constants.py (hunk 1 above)
Comment (amandate) [reply src/Constants/constants.py]: Done!

**Review by leezahuor — APPROVED** [review]
Thank you for the explanations! Everything looks good!
